Gulpfile.js

```diff
@@ -10,41 +10,39 @@ const gutil = require("gulp-util");
 const gulp = require("gulp");
 const path = require("path");
 
+const base = path.join(__dirname, "packages");
 const scripts = "./packages/*/src/**/*.js";
-let srcEx, libFragment;
 
-if (path.win32 === path) {
-  srcEx = /(packages\\[^\\]+)\\src\\/;
-  libFragment = "$1\\lib\\";
-} else {
-  srcEx = new RegExp("(packages/[^/]+)/src/");
-  libFragment = "$1/lib/";
+function swapSrcWithLib(srcPath) {
+  const parts = srcPath.split(path.sep);
+  parts[1] = "lib";
+  return parts.join(path.sep);
 }
 
-const mapToDest = function (path) { return path.replace(srcEx, libFragment); };
-const dest = "packages";
-
 gulp.task("default", ["build"]);
 
 gulp.task("build", function () {
-  return gulp.src(scripts)
+  return gulp.src(scripts, { base: base })
```

Comment:
This was the core of the problem, we weren't setting a base, so all the filepaths were `babel-core/src/index.js`, and the regexes all expected the path to start with `packages`.